app/src/main/java/com/io/movio/fragments/MovieDetailFragment.kt

```diff
@@ -17,28 +17,26 @@ class MovieDetailFragment : Fragment() {
     private lateinit var binding: FragmentMovieDetailBinding
     private lateinit var movie: Movie
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            movie = it.getSerializable(ARG_MOVIE) as Movie
-        }
-    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.apply {
-            binding.tvTitle.text = movie.title
-            binding.tvCast.text = movie.cast
-            binding.tvDescription.text = movie.description
-            binding.tvDescription.movementMethod = ScrollingMovementMethod()
+            tvTitle.text = movie.title
+            tvCast.text = movie.cast
+            tvDescription.text = movie.description
+            tvDescription.movementMethod = ScrollingMovementMethod()
         }
     }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?): View {
+
         binding = FragmentMovieDetailBinding.inflate(inflater , container , false)
 
+        arguments?.let {
```

Comment:
here you can also use require arguments BUT ONLY IF your screen shouldn't be working without such provided. In this way you can catch your errors while developing because it will crash and will avoid weird behavior.